mesonbuild/build.py

```diff
@@ -721,6 +721,9 @@ class BuildTarget(Target):
         ('cpp', 'cuda'),
     ])
 
+    # This is used by the backend to cache complex computation.
+    cached_generated_headers: T.Optional[T.List[FileOrString]]
```

Comment:
I have some patches somewhere to stop adding this to the Target and just store it in the NijnjaBackend. I'm fine with this being here for the short term, but it's a really ugly hack and we should not do it in the long run.

Reply:
Yup, it's incredibly ugly as can also be seen by the fact that hasattr is ever used at all. :D